tests/test_raster.py

```diff
@@ -2966,51 +2984,122 @@ def test_proximity_parameters(self) -> None:
         # With only inside proximity
         raster1.proximity(vector=vector, in_or_out="in")
 
-    def test_to_points(self) -> None:
-        """Test the outputs of the to_points method and that it doesn't load if not needed."""
+    def test_to_pointcloud(self) -> None:
+        """Test to_pointcloud method."""
+
+        # 1/ Single band synthetic data
 
         # Create a small raster to test point sampling on
-        img0 = gu.Raster.from_array(
-            np.arange(25, dtype="int32").reshape(5, 5), transform=rio.transform.from_origin(0, 5, 1, 1), crs=4326
-        )
+        img_arr = np.arange(25, dtype="int32").reshape(5, 5)
+        img0 = gu.Raster.from_array(img_arr, transform=rio.transform.from_origin(0, 5, 1, 1), crs=4326)
 
         # Sample the whole raster (fraction==1)
-        points = img0.to_points(1, as_array=True)
+        points = img0.to_pointcloud()
+        points_arr = img0.to_pointcloud(as_array=True)
 
-        # Validate that 25 points were sampled (equating to img1.height * img1.width) with x, y, and band0 values.
-        assert isinstance(points, np.ndarray)
-        assert points.shape == (25, 3)
-        assert np.array_equal(np.asarray(points[:, 0]), np.tile(np.linspace(0.5, 4.5, 5), 5))
+        # Check output types
+        assert isinstance(points, gu.Vector)
+        assert isinstance(points_arr, np.ndarray)
 
-        assert img0.to_points(0.2, as_array=True).shape == (5, 3)
+        # Check that both outputs (array or vector) are fully consistent, order matters here
+        assert np.array_equal(points.ds.geometry.x.values, points_arr[:, 0])
+        assert np.array_equal(points.ds.geometry.y.values, points_arr[:, 1])
+        assert np.array_equal(points.ds["b1"].values, points_arr[:, 2])
+
+        # Validate that 25 points were sampled (equating to img1.height * img1.width) with x, y, and band0 values.
+        assert isinstance(points_arr, np.ndarray)
+        assert points_arr.shape == (25, 3)
+        assert points.ds.shape == (25, 2)  # One less column here due to geometry storing X and Y
+        # Check that X, Y and Z arrays are equal to raster array input independently of value order
+        x_coords, y_coords = img0.ij2xy(i=np.arange(0, 5), j=np.arange(0, 5))
```

Comment:
can't you just use `Raster.coords`? https://geoutils.readthedocs.io/en/latest/gen_modules/geoutils.Raster.coords.html

Reply:
No as it didn't use ij2xy consistently yet, this was corrected in #509. I take a note and I'll see if it makes sense to change when merging conflicts.